### ISSUE TITLE
Repeating Same Artists & Songs

### ISSUE THREAD

**Issue by djanthonystyles** [AzuraCast/AzuraCast#2739]
**Using Docker installation method**
Yes

**Host Operating System**
Ubuntu 18.04.4 LTS / 4 VCPU /16 GB RAM / 280 GB HD

**Describe the bug**
The same artists and songs keep playing within a short time frame after updating to the most recent version of AzuraCast. Won't play other playlists like rare rotation and low rotation. Just keeps playing from Power Hit and Heavy Rotation. I've tried restarting the station, which includes stopping it in the terminal and starting it back up. I also tried deleting AzuraCast and starting from scratch with a fresh installation. The issue stayed, before the update, it worked flawlessly.

**To Reproduce**

**Expected behavior**
Expecting the same artist and songs to be spread out more evenly.

**Relevant Logs**

**Screenshots**
![viH1CJ](https://user-images.githubusercontent.com/64068105/80281936-96d9ba80-86dc-11ea-8c56-8a8a53997401.png)
![qvOCxN](https://user-images.githubusercontent.com/64068105/80282167-f2f10e80-86dd-11ea-8a88-f5c2b7255ea2.png)

![R4T1a6](https://user-images.githubusercontent.com/64068105/80281976-cf799400-86dc-11ea-9ef8-daa100044f7d.png)
![rjl5CO](https://user-images.githubusercontent.com/64068105/80282005-094a9a80-86dd-11ea-87b5-cea38c9f93c3.png)

**Device(s):**
 - Device:  
 - OS: 
 - Browser: 
 - Version: 

**Additional context**

**Comment by djanthonystyles** [comment]
Been messing with it, and the only way to get other playlists to work is by making them all the same weight number. Otherwise, the higher number will play back to back and lower number playlists don't get played at all.

**Comment by djanthonystyles** [comment]
I also wanted to add looking docker-compose logs showing this constantly after updating to latest AzuraCast version. web_1       | sudo: setrlimit(RLIMIT_CORE): Operation not permitted

**Comment by SlvrEagle23** [comment]
@djanthonystyles The second issue is an unrelated one to the scheduling problems.

I've found the problem with weighted playlists and duplicate prevention, and this should now be fixed as of the latest update. Give it a few minutes, then update and make sure you are on commit `b608b2` (or newer, if reading this after the fact) to have the fixes applied.

**Comment by SlvrEagle23** [comment]
Following a Discord discussion, it appears this issue is still happening for the user in question. However, they've switched to "Manual AutoDJ" mode to resolve it, which removes the AzuraCast AutoDJ from the equation and replaces it with Liquidsoap's internal AutoDJ, so with that setting enabled, there is no further testing that we can do on that installation.

If you're experiencing this issue still and using the AzuraCast AutoDJ, please feel free to create a new issue with further information.

**Comment by djanthonystyles** [comment]
I tried both ways and still have the same issue. Setting playlists to the same weight will allow them to play, but kinda defeats the purpose of weighted rotation. Just won't play any of the lower weighted playlists.  I prefer to use AzuraCast AutoDJ. Thanks

**Comment by SlvrEagle23** [comment]
@djanthonystyles I've made further changes to this shuffling algorithm, so you can update again and see if it fixes the issue now.

**Comment by djanthonystyles** [comment]
@SlvrEagle23 Update seems to let other playlists get placed into the rotation now, evenly. But, artists playing back to back or close to each other.


Apr 27, 2020 12:56 PM | 15 | trending_up 4 | La Romana (Ft. El Alfa) - Bad Bunny | Playlist: Light Rotation
-- | -- | -- | -- | --
Apr 27, 2020 12:53 PM | 9 | trending_up 6 | Yo Perreo Sola - Bad Bunny | Playlist: Medium Rotation

**Comment by djanthonystyles** [comment]
When hitting debug of a station noticed that it now rebuilds the autodj. Does it movie songs around to other playlists when this happens? Noticed most of the songs get moved around, even though auto-assignment isn't for another week.

**Comment by sr8087as** [comment]
I've got the latest version & still seeing back-to-back artist plays occasionally

**Comment by pauler-justin** [comment]
I am glad I am not alone here! I too have issues with artists/albums playing back-to-back - enough that listeners have reached out to ask if there was a problem. In some cases, an artist will play 2 or 3 times in a row, which as I understand it, is a DMCA violation (I am not a lawyer, and do not play one on TV):

https://support.live365.com/hc/en-us/articles/115002892247-What-is-DMCA-

In an effort to curb some of the repeats, I wrote the following code and added it to my Liquidsoap configuration from within AzuraCast. This is by no means perfect, and is subject to change - however, it has proven to help. Use at your own risk!

https://pastebin.pl/view/b5ca8eb5

**Comment by djanthonystyles** [comment]
@sr8087as @jhp45 Sorry to hear that, I regret updating. @jhp45 Has the code been working for you? Thinking of trying this out. Thanks

**Comment by djanthonystyles** [comment]
I'll give it a shot tonight because it's really bad for me. It just played the same song again but showing it in another playlist. So confused.

May 1, 2020 4:43 AM | 626 | trending_up 5 | Lo Que Paso PasoDaddy Yankee | Playlist: Light Rotation
-- | -- | -- | -- | --
May 1, 2020 4:40 AM | 610 | trending_up 16 | Lo Que Paso, PasoDaddy Yankee | Playlist: Heavy Rotation

**Comment by SlvrEagle23** [comment]
> Sorry to hear that, I regret updating

@djanthonystyles Don't say things like this. You're invalidating a lot of hard work that's been done to help resolve bugs and solve problems.

Our duplicate prevention is reasonably strict, and it will avoid duplicates where it absolutely can, but if there is only one playlist of one type available to play, it will still play something from that playlist even if a duplicate.

I'm considering making this a per-playlist setting, since some playlists simply don't require any duplicate prevention (i.e. sweepers), while others you may find yourself wanting the matching to be so strict that it simply would prefer to skip the playlist entirely and not play anything if it can't avoid a duplicate.

As always (and I feel like I say this in vain these days, but I'll say it anyway): AzuraCast is open-source and we welcome pull requests from the community if you have fixes for critical functionality. Anyone in the whole wide world can contribute to AzuraCast, so it doesn't need to just be me.

**Comment by djanthonystyles** [comment]
@SlvrEagle23 Sorry, let me rephrase that. I regret not backing up my install, which is my fault. Wasn't a jab at anyone or the project. I understand it's open-source that's why I'm here trying to report bugs and interact with other users having issues to find a solution.

Not trying to play down any of the hard work you put in, I appreciate everything you have done. Hence why I'm donating monthly to help with the project.

That could work, ill look forward it. Thanks

**Comment by o0OcyberjunkieO0o** [comment]
Hello there. 
I'm having the same problems, and since my radio is about electronic music, repeating them is really bad. I have more than 24gb of music and it rotates so bad on the shuffle mode.
Please help us <3
